pkg/deploy/api/v1/types.go

```diff
@@ -370,3 +370,31 @@ type DeploymentLogOptions struct {
 	// Version of the deploymentConfig for which to view logs.
 	Version *int `json:"version,omitempty" description:"the version of the deploymentConfig for which to view logs"`
 }
+
+// ScaleSpec describes the attributes a Scale subresource
+type ScaleSpec struct {
```

Comment:
There was weirdness and bad error messages.  Can you try https://github.com/openshift/origin/pull/5364 and see if you still need this?  I'd prefer to avoid having different scale objects if we can avoid it.  It might not be possible because deploymentconfigs are under `oapi/v1`, but I'd like to attempt it.